core/src/main/scala/org/apache/spark/sql/execution/CodegenSparkFallback.scala

```diff
@@ -87,6 +88,15 @@ case class CodegenSparkFallback(var child: SparkPlan) extends UnaryExecNode {
             }
           case None => throw t
         }
+    } finally {
+      // SNAP-1422
+      val cacheMetaDataForConnector =
+        java.lang.Boolean.getBoolean("SMART_CONNECTOR_CACHE_METADATA")
+      lazy val session = sqlContext.sparkSession.asInstanceOf[SnappySession]
+      if (!cacheMetaDataForConnector &&
+        ExternalStoreUtils.isSmartConnectorMode(session.sparkContext)) {
+        session.sessionCatalog.invalidateAll()
```

Comment:
Ok, ignore last comment. Looked at the bug again and this case is when tableExists is incorrectly returning true. I think we should just document that particular case and provide a public API to clear the meta-data explicitly (or user can create a new session). Not worth it to take the hit of clearing the whole cache every time just for that case IMO.